### ISSUE TITLE
Add the ability to create simulated packets

### ISSUE THREAD

**Issue by ehsteve** [ddasilva/ccsdspy#50]
It is often very helpful to be able to create simulated packets especially in the early phases of development or for testing. The packet types `FixedLength` and `VariableLength` could be expanded to include a function (`to_file`?) to output binary data if given arrays of data to fill in the packet fields.

```python
to_file(data: dict)
```
where `dict` includes the packet field names as indices and numpy arrays for the values. The data dict should mirror the data dict that is provided by parsing the file.

**Comment by ehsteve** [comment]
This issue is being worked in the [packet_encoding](https://github.com/CCSDSPy/ccsdspy/tree/packet_encoding) branch. Basic functionality is already implemented and ready to be tested by anyone interested!

**Comment by ddasilva** [comment]
I'm open to this-- it really has no impact to existing users and it would be pretty nice to generate pretty HTML. Having the definitions expressed in python might not be the best authoritative source for the packet definitions in a large project, but if they're already in Python its is a nice bonus feature.

What do you think @ehsteve ? 
Would this be useful for you @jmbhughes? Maybe PUNCH could release nicely formatted HTML documentation of the packet definitions.

**Comment by jmbhughes** [comment]
At the moment, we are just reading the definitions from a CSV file and were going to distribute that, but it might be nice to have them presented nicely in the website. If it was pretty and didn't require too much additional work, we'd use it probably. 

We also may try out this simulated packet creation to test our pipeline soon.

**Comment by ddasilva** [comment]
Yes! If you validate this works and log any remaining issues that will help a lot. It probably has a few merge conflicts since it was forked a while ago.
